RecoLocalCalo/HcalRecAlgos/src/HcalDeterministicFit.cc

```diff
@@ -3,6 +3,10 @@
 #include <climits>
 #include "RecoLocalCalo/HcalRecAlgos/interface/HcalDeterministicFit.h"
 
+const float HcalDeterministicFit::invGpar[3];
+const float HcalDeterministicFit::negthre[2];
+const int HcalDeterministicFit::HcalRegion[2];
+
```

Comment:
Depends on their usage. C++ allows to define and initialize variables in class/struct, but if it's odr-used in close it must generate the symbol. In that case standard says that compiler diagnostic is not required IIRC, i.e., symbol not generated, linker is not happy. I don't know if constexpr gives an extra play here.

Read C++11 standard 9.4.2/3 and 9.4.2/4. 

Or C++14 (as we are using it now): http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf

Sames sections.